tools/common/CompilerFlags.cs

```diff
@@ -19,13 +19,32 @@ public class CompilerFlags
 		public HashSet<string> OtherFlags; // X
 		public HashSet<string> Defines; // -DX
 		public HashSet<string> UnresolvedSymbols; // -u X
+		public HashSet<string> SourceFiles; // X, added to Inputs
 
 		// Here we store a list of all the file-system based inputs
 		// to the compiler. This is used when determining if the
 		// compiler needs to be called in the first place (dependency
 		// tracking).
 		public List<string> Inputs;
 
+		public CompilerFlags (Target target)
+		{
+			if (target == null)
+				throw new ArgumentNullException (nameof (target));
+			this.Target = target;
+		}
+
+		public HashSet<string> AllLibraries {
+			get {
+				var rv = new HashSet<string> ();
```

Comment:
The property is only fetched once, but I can make it a method instead to make it clearer there's a significant cost to calling it.